hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -286,19 +286,20 @@ public static Map<String, HoodieColumnRangeMetadata<Comparable>> collectColumnRa
       // with the values from this record
       targetFields.forEach(fieldNameFieldPair -> {
         String fieldName = fieldNameFieldPair.getKey();
-        Schema fieldSchema = getNonNullTypeFromUnion(fieldNameFieldPair.getValue().schema());
-        ColumnStats colStats = allColumnStats.computeIfAbsent(fieldName, ignored -> new ColumnStats(getValueMetadata(fieldSchema, indexVersion)));
-        Object fieldValue = collectColumnRangeFieldValue(record, colStats.valueMetadata, fieldName, fieldSchema, recordSchema, properties);
+        HoodieSchemaField field = fieldNameFieldPair.getValue();
+        HoodieSchema hoodieFieldSchema = HoodieSchemaUtils.getNonNullTypeFromUnion(field.schema());
```

Comment:
nitpick: can we keep the variable name `fieldSchema`?

Reply:
Done